xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java

```diff
@@ -34,21 +34,22 @@ final class WeightedRandomPicker extends SubchannelPicker {
   final List<WeightedChildPicker> weightedChildPickers;
 
   private final ThreadSafeRandom random;
-  private final int totalWeight;
+  private final long totalWeight;
 
   static final class WeightedChildPicker {
-    private final int weight;
+    private final long weight;
     private final SubchannelPicker childPicker;
 
-    WeightedChildPicker(int weight, SubchannelPicker childPicker) {
+    WeightedChildPicker(long weight, SubchannelPicker childPicker) {
       checkArgument(weight >= 0, "weight is negative");
+      checkArgument(weight <= 2L * Integer.MAX_VALUE, "weight is too large");
```

Comment:
max unsigned int is 2L * Integer.MAX_VALUE + 1. 
Use UnsignedInteger.MAX_VALUE.longValue()?

Reply:
done.